### PR TITLE
Writing the returned progress description from the broker

### PR DESCRIPTION
**Pull request by arschles**
Fixes https://github.com/kubernetes-incubator/service-catalog/issues/997

### CODE REVIEW

File: pkg/controller/controller_instance.go [hunk 1]
```diff
@@ -460,9 +460,31 @@ func (c *controller) pollInstance(serviceClass *v1alpha1.ServiceClass, servicePl
...
+				return err
+			}
+			toUpdate := clone.(*v1alpha1.Instance)
+			toUpdate.Status.AsyncOpInProgress = true
+
+			reason := asyncProvisioningReason
+			message := fmt.Sprintf("%s (%s)", asyncProvisioningMessage, *response.Description)
+			if deleting {
+				reason = asyncDeprovisioningReason
+				message = fmt.Sprintf("%s (%s)", asyncProvisioningMessage, *response.Description)
```
Comment (MHBauer) [inline pkg/controller/controller_instance.go]: Provisioning or deprovisioning message?

I don't really like this construct of 'set the variables, and if we happen to need to, overwrite them. Can we get an if else for each specific case instead?

File: pkg/controller/controller_instance_test.go [hunk 2]
```diff
@@ -892,8 +900,10 @@ func TestPollServiceInstanceInProgressDeprovisioningWithOperationNoFinalizer(t *
 		t.Fatalf("pollInstanceInternal failed but not with expected error, expected %q got %q", "still in progress", err)
 	}
 
+	// there should have been 1 action to update the instance status with the last operation
+	// description
 	actions := fakeCatalogClient.Actions()
-	assertNumberOfActions(t, actions, 0)
+	assertNumberOfActions(t, actions, 1)
```
Comment (MHBauer) [inline pkg/controller/controller_instance_test.go]: Can we check what the action is ? To make sure it is the right one?

File: pkg/controller/controller_instance.go (hunk 1 above)
Comment (MHBauer) [inline pkg/controller/controller_instance.go]: Is there something to add to the test to make sure it's getting the right reason in the update condition?

File: pkg/controller/controller_instance_test.go (hunk 2 above)
Comment (nilebox) [inline pkg/controller/controller_instance_test.go]: We do, see for example https://github.com/kubernetes-incubator/service-catalog/pull/957/files#diff-34dbc217e78a01dffcd4dfcd5d973d3cR392

File: pkg/controller/controller_instance.go [hunk 3]
```diff
@@ -460,9 +460,31 @@ func (c *controller) pollInstance(serviceClass *v1alpha1.ServiceClass, servicePl
...
+			// we need to keep on polling.
+			clone, err := api.Scheme.DeepCopy(instance)
+			if err != nil {
+				return err
+			}
+			toUpdate := clone.(*v1alpha1.Instance)
+			toUpdate.Status.AsyncOpInProgress = true
+
+			reason := asyncProvisioningReason
+			message := fmt.Sprintf("%s (%s)", asyncProvisioningMessage, *response.Description)
```
Comment (pmorie) [inline pkg/controller/controller_instance.go]: This will panic if description is nil

File: pkg/controller/controller_instance_test.go (hunk 2 above)
Comment (arschles) [inline pkg/controller/controller_instance_test.go]: @MHBauer done

**Comment by arschles** [comment]
@MHBauer updates made to address your command, can you please take another look?

@pmorie I only found one dereference of `response.Description` in this PR, and added a nil check around it per your comment. Were there others you saw?

**Comment by arschles** [comment]
@MHBauer will do

**Comment by pmorie** [comment]
looks good now!

**Pull request merged**